uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -225,6 +225,75 @@ extension Bool: ViaFfi {
     }
 }
 
+extension Date: ViaFfiUsingByteBuffer, ViaFfi {
+    fileprivate static func read(from buf: Reader) throws -> Self {
+        let seconds: Int64 = try buf.readInt()
+        let microseconds: UInt32 = try buf.readInt()
+        if seconds >= 0 {
+            let delta = Double(seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: delta)
+        } else {
+            let delta = Double(-seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: -delta)
+        }
+    }
+
+    fileprivate func write(into buf: Writer) {
+        var delta = self.timeIntervalSince1970
+        var sign: Int64 = 1
+        if delta < 0 {
+            sign = -1
+            delta = -delta
+        }
+        if delta.rounded(.down) > Double(Int64.max) {
+            fatalError("Timestamp overflow, exceeds max bounds supported by Uniffi")
+        }
+        let seconds = Int64(delta)
+        let microseconds = UInt32((delta - Double(seconds)) * 1.0e9)
+        buf.writeInt(sign * seconds)
```

Comment:
Similarly here, I don't understand why we need to extract `sign` from `delta` and the multiple them back together. Is it so that we can do the microseconds calculation more easily? Please consider adding a comment to help future readers (aka myself in 6 months from now when I forget what this was all about 😅)

Reply:
Added a comment a few lines up